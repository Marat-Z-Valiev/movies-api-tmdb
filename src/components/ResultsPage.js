import React, {useState, useEffect} from "react";
import {withRouter} from "react-router-dom";
import axios from "axios";
import Results from "./Results";
import PaginationComponent from "./PaginationComponent";

const apiKey = "f8efee7e451d2ca98ae50114ad74aeeb";

const baseUrl = "https://api.themoviedb.org/3/search/movie?api_key=";

const ResultsPage = ({match}) => {
	const queryValue = match.params.searchQuery;
	console.log(queryValue);
	let [results, setResults] = useState([]);
	let [totalResults, setTotalResults] = useState(0);
	let [isError, setIsError] = useState(false);
	let [isLoading, setIsLoading] = useState(false);
	let [isDisabled, setIsDisabled] = useState(true);
	let [showPagination, setShowPagination] = useState(false);
	let [currentPage, setCurrentPage] = useState(1);
	const [resultsPerPage] = useState(10);

	const getResults = async currentPage => {
		await axios
			.get(
				`${baseUrl}${apiKey}&language=en-US&query=${queryValue}&page=${currentPage}`
			)
			.then(response => {
				if (response.data.results.length) {
					setIsError((isError = false));
					setResults((results = response.data.results));
					setTotalResults((totalResults = response.data.total_results));
					setIsLoading((isLoading = false));
					setShowPagination((showPagination = true));
				} else {
					setIsError((isError = true));
					setIsLoading((isLoading = false));
					setShowPagination((showPagination = false));
				}
			})
			.catch(err => console.log(`this is error ${err}`));
	};

	useEffect(() => {
		getResults();
	}, [queryValue]);

	return (
		<>
			<h2>Results page</h2>
			{/* {showPagination ? (
				<PaginationComponent
					currentPage={currentPage}
					handlePageChange={handlePageChange}
					resultsPerPage={resultsPerPage}
					totalResults={totalResults}
				/>
			) : (
				""
			)} */}
			{isError ? (
				<p className="error-message">No Results Found</p>
			) : (
				<Results
					results={results}
					totalResults={totalResults}
					isError={isError}
				/>
			)}
			{/* {showPagination ? (
				<PaginationComponent
					currentPage={currentPage}
					handlePageChange={handlePageChange}
					resultsPerPage={resultsPerPage}
					totalResults={totalResults}
				/>
			) : (
				""
			)} */}
		</>
	);
};

export default withRouter(ResultsPage);
