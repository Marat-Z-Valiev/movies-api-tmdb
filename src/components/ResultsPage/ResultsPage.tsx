import React, {useState, useEffect} from "react";
import {withRouter, RouteComponentProps} from "react-router-dom";
import axios from "axios";
import PropTypes from "prop-types";
import Results from "../Results/Results";
import PaginationComponent from "../Pagination/PaginationComponent";

const apiKey = process.env.API_KEY;

const baseUrl = "https://api.themoviedb.org/3/search/movie?api_key=";

interface ResultsPageProps {
	match: any;
}

const ResultsPage: React.FC<ResultsPageProps & RouteComponentProps> = ({
	match,
}) => {
	const queryValue = match.params.searchQuery;
	let [results, setResults] = useState([]);
	let [totalResults, setTotalResults] = useState(0);
	let [isError, setIsError] = useState(false);
	let [showPagination, setShowPagination] = useState(false);
	let [currentPage, setCurrentPage] = useState(1);
	const [resultsPerPage] = useState(10);

	const getResults = async (queryValue: string, currentPage: number) => {
		await axios
			.get(
				`${baseUrl}${apiKey}&language=en-US&query=${queryValue}&page=${currentPage}`
			)
			.then((response) => {
				if (response.data.results.length) {
					setIsError((isError = false));
					setResults((results = response.data.results));
					setTotalResults((totalResults = response.data.total_results));
					setShowPagination((showPagination = true));
				} else {
					setIsError((isError = true));
					setShowPagination((showPagination = false));
				}
			})
			.catch((err) => console.log(`this is error ${err}`));
	};

	useEffect(() => {
		getResults(queryValue, currentPage);
	}, [queryValue]);

	const handlePageChange = (pageNumber: number): void => {
		setCurrentPage((currentPage = pageNumber));
		getResults(queryValue, pageNumber);
	};

	return (
		<>
			<h2 style={{textAlign: "center", color: "#ffffff"}}>Results page</h2>
			<h2 style={{textAlign: "center"}}>
				{`Showing results for "${queryValue}"`}
			</h2>
			{showPagination ? (
				<PaginationComponent
					currentPage={currentPage}
					handlePageChange={handlePageChange}
					resultsPerPage={resultsPerPage}
					totalResults={totalResults}
				/>
			) : (
				""
			)}
			{isError ? (
				<p className="error-message">No Results Found</p>
			) : (
				<Results
					results={results}
					totalResults={totalResults}
					isError={isError}
				/>
			)}
			{showPagination ? (
				<PaginationComponent
					currentPage={currentPage}
					handlePageChange={handlePageChange}
					resultsPerPage={resultsPerPage}
					totalResults={totalResults}
				/>
			) : (
				""
			)}
		</>
	);
};

ResultsPage.propTypes = {
	match: PropTypes.object.isRequired,
};

export default withRouter(ResultsPage);
