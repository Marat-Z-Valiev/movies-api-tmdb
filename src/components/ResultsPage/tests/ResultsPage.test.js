import React from "react";
import {shallow, configure} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import ResultsPage from "../ResultsPage";

configure({adapter: new Adapter()});

describe("ResultsPage", () => {
	it("should render correctly", () => {
		const props = {
			match: {
				params: {
					searchQuery: "test",
				},
			},
		};
		const wrapper = shallow(<ResultsPage {...props} />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
