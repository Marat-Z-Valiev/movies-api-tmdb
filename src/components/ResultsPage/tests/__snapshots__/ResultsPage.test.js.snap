// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsPage should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
